hello.py

```diff
@@ -1,33 +1,69 @@
-from flask import Flask, render_template
+from flask import Flask, render_template, jsonify
 from flask_nav import Nav
-from flask_nav.elements import Navbar, View
+from flask_nav.elements import Navbar, View, Subgroup
+import json
 
 app = Flask(__name__)
 nav = Nav(app)
 
-nav.register_element("top", Navbar(
+with open("database.json", "r") as d:
+  data = json.load(d)
+
+categories = sorted(set([item["category"] for item in data]))
+categories_sub = Subgroup("Categories")
+
+for cat in categories:
+    categories_sub.items.append(View(cat, "category", cat = cat))
+
+topbar = Navbar(
     "Top_bar",
     View("Home", "index"),
-    View("Categories", "category"),
+    View("All Products", "all_cats"),
+    categories_sub,
     View("About Mobius", "about")
-))
+)
+
+nav.register_element("top", topbar)
 
 @app.route("/")
 def index():
     return render_template("index.html")
 
-@app.route("/product")
-def product():
-    return render_template("product.html")
+@app.route("/product/<name>")
+def product(name):
+    item_data = get_item_from_data(name, data)
+    return render_template("product.html", 
+                           name = item_data["name"], 
+                           price = item_data["price"], 
+                           desc = item_data["desc"], 
+                           img1 = item_data["images"][0],
+                           img2 = item_data["images"][1])
 
 @app.route("/category")
-def category():
-    return render_template("categories.html")
+def all_cats():
+    return render_template("show_all.html", data = data)
 
+@app.route("/category/<cat>")
+def category(cat):
+    cat_items = items_by_cat(cat, data)
+    return render_template("category.html", cat_items = cat_items)
+    
 @app.route("/about")
 def about():
     return render_template("about.html")
 
+def items_by_cat(cat, data):
```

Comment:
A more optimal way of doing this would be to categorise items in advance at the point where you are parsing the JSON, so that you don't need to do a search every time someone visits the `/category/<cat>` endpoint. You could have `categories = {"clothes": [1, 4, 6]}` where the numbers are indexes into the items list from the JSON. Then all you need to get the item would be `data[categories[cat]]`.

However, what you have here is totally fine for this stage as we'll be replacing this lookup code with some database model.